src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,193 @@
 import java.awt.Graphics2D;
 import java.awt.Image;
 import java.awt.Point;
-import java.awt.Polygon;
 import java.awt.RenderingHints;
-import java.awt.Shape;
-import java.awt.geom.AffineTransform;
-import java.awt.geom.Ellipse2D;
 import java.awt.geom.Line2D;
 import java.awt.geom.Point2D;
-import java.awt.geom.QuadCurve2D;
 import java.awt.image.BufferedImage;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
+import org.apache.commons.math3.analysis.interpolation.SplineInterpolator;
+import org.apache.commons.math3.analysis.polynomials.PolynomialSplineFunction;
+
 import games.strategy.engine.data.Route;
 import games.strategy.engine.data.Territory;
+import games.strategy.util.Tuple;
 
 /**
- * Draws a route on a map. This code is really ugly, bad and it barely works. It
- * should be rewritten.
+ * Draws a route on a map.
  */
 public class MapRouteDrawer {
-  // only static methods
-  private MapRouteDrawer() {}
+
+  private static final SplineInterpolator splineInterpolator = new SplineInterpolator();
 
   /**
-   * Draw m_route to the screen, do nothing if null.
+   * Draws the route to the screen, does nothing if null.
```

Comment:
I think the route is meant... Kept this comment from the original route drawer...